src/ray/object_manager/pull_manager.cc

```diff
@@ -461,15 +461,15 @@ void PullManager::TryToMakeObjectLocal(const ObjectID &object_id) {
   }
 
   // If we can restore directly from this raylet, then try to do so.
-  std::string spilled_url = get_locally_spilled_object_url_(object_id);
+  const std::string spilled_url = get_locally_spilled_object_url_(object_id);
```

Comment:
Thanks for catching this bug! 👍 should we fallback to request.spilled_url if local spilled is empty otherwise it doesn't work with S3? 

also is this possible to add unit test for this?

Reply:
yep sure. Will do tmrw